test/sparsedir/sparsevector.jl

```diff
@@ -762,3 +763,12 @@ a14013 = [10.0 0.0 30.0; 0.0 1.0 0.0]
 s14046 = sprand(5, 1.0)
 @test spzeros(5) + s14046 == s14046
 @test 2*s14046 == s14046 + s14046
+
+# Issue 14589
+# test that sorting behaves the same as dense vectors
+# apply three distinct tranformations where zeros sort into start/middle/end
+x = sparsevec(1:7, [3, 2, -1, 1, -2, -3, 3], 15)
```

Comment:
Would also be good to add a test where the nonzero indices are `1:length(x)` so there is actually no implicit zero element present. I think your implementation works for this case but it's not immediately obvious.